ui/components/EventMembers/index.tsx

```diff
@@ -135,6 +126,14 @@ const EventMembers = ({ collection, currentUser }) => {
     pause: true,
   });
 
+  useEffect(() => {
```

Comment:
Can we simply have simple variables instead of extra react state for this? E.g. `const approvedMembers = data?.approvedMembersPage?.approvedMembers || [];`

Reply:
Yes, we can do it. This approach is much simpler. 

This way i think there is a small advantage. For example, is moreExists is false, we can search(filter) from local data instead of calling api.(right now api is called).

We may want to implement pagination in future. So if we add pagination, `const approvedMembers = data?.approvedMembersPage?.approvedMembers || [];` will only return limited numbers of result.

For example, 10 results are loaded on page load. When the window is scrolled down, 10 more results are fetched. In this case, `const approvedMembers = data?.approvedMembersPage?.approvedMembers || [];` will override the previously loaded data.

By using a separate state, we can append the newly added loaded data to the state.

Reply:
I think @Powersource's suggestion is the right one, it seems simpler to me.

Also, currently the strategy to do pagination is to add another Page component, and not append individual items to an array in state, see the buckets list for example.

And also, I think searching should be done by the API as it is now, it's more likely that we want to search the entire database records and not only filter the items currently displayed on the page.